File: src/GestorDeTareas.Application/DTOs/ActualizarTareaDto.cs
using GestorDeTareas.Domain.Enums;

namespace GestorDeTareas.Application.DTOs;

public class ActualizarTareaDto
{
    public string Titulo { get; set; } = string.Empty;
    public DateTime FechaLimite { get; set; }
    public PrioridadTarea Prioridad { get; set; }
    public EstadoTarea Estado { get; set; }
    public PilarVida Pilar { get; set; }

    public string? Intencion { get; set; }
}

File: gestor-tareas-frontend/src/app/features/tareas/models/actualizar-tarea.dto.ts
import { EstadoTarea } from './estado-tarea.enum';
import { PilarVida } from './pilar-vida.enum';
import { PrioridadTarea } from './prioridad-tarea.enum';

export interface ActualizarTareaDto {
  titulo: string;
  fechaLimite: string;
  prioridad: PrioridadTarea;
  estado: EstadoTarea;
  pilar: PilarVida;
  intencion?: string | null;
}